internal/ports/vlans.go

```diff
@@ -50,28 +51,35 @@ func (c *Client) Vlans() *cobra.Command {
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
-			listOpts := c.Servicer.ListOptions([]string{"port"}, nil)
+			// inc := []string{"port"}
+			// exc := []string{}
+
+			req := metal.NewPortVlanAssignmentBatchCreateInput()
 
-			getOpts := &packngo.GetOptions{Includes: listOpts.Includes, Excludes: listOpts.Excludes}
-			req := &packngo.VLANAssignmentBatchCreateRequest{}
 			f := false
 			t := true
 			for _, vlan := range assignments {
-				assignment := packngo.VLANAssignmentCreateRequest{VLAN: vlan, State: packngo.VLANAssignmentAssigned, Native: &f}
-				req.VLANAssignments = append(req.VLANAssignments, assignment)
+				CreateInputVlanAssignmentsInne := []metal.PortVlanAssignmentBatchCreateInputVlanAssignmentsInner{}
+				CreateInputVlanAssignmentsInne[0].SetVlan(vlan)
+				CreateInputVlanAssignmentsInne[0].SetNative(f)
+				req.SetVlanAssignments(CreateInputVlanAssignmentsInne)
 			}
 			for _, vlan := range unassignments {
-				assignment := packngo.VLANAssignmentCreateRequest{VLAN: vlan, State: packngo.VLANAssignmentUnassigned}
-				req.VLANAssignments = append(req.VLANAssignments, assignment)
+				CreateInputVlanAssignmentsInne := []metal.PortVlanAssignmentBatchCreateInputVlanAssignmentsInner{}
+				CreateInputVlanAssignmentsInne[0].SetVlan(vlan)
+				req.SetVlanAssignments(CreateInputVlanAssignmentsInne)
 			}
 			if native != "" {
-				assignment := packngo.VLANAssignmentCreateRequest{VLAN: native, State: packngo.VLANAssignmentAssigned, Native: &t}
-				req.VLANAssignments = append(req.VLANAssignments, assignment)
+				CreateInputVlanAssignmentsInner := []metal.PortVlanAssignmentBatchCreateInputVlanAssignmentsInner{}
+				CreateInputVlanAssignmentsInner[0].SetVlan(native)
+				CreateInputVlanAssignmentsInner[0].SetNative(t)
+				req.SetVlanAssignments(CreateInputVlanAssignmentsInner)
 			}
-			if len(req.VLANAssignments) == 0 {
+
+			if len(req.GetVlanAssignments()) == 0 {
 				return errors.New("no VLAN assignments specified")
 			}
-			batch, _, err := c.VLANService.CreateBatch(portID, req, getOpts)
+			batch, _, err := c.PortService.CreatePortVlanAssignmentBatch(context.Background(), portID).PortVlanAssignmentBatchCreateInput(*req).I
```

Comment:
```suggestion
			batch, _, err := c.PortService.CreatePortVlanAssignmentBatch(context.Background(), portID).PortVlanAssignmentBatchCreateInput(*req).Execute()
```

Reply:
This needs to be addressed in spec side first misssing Includes, Excludes

Reply:
Any issues like this where you are aware that the `metal-go` client is missing features that are needed for conversion should be noted on the relevant PR and the PR should be placed in draft status.  That will protect against accidental merge of changes that are known to be not completely working, and will make it easier for the whole team to understand what issues are blocking the remaining `metal-go` conversion PRs.